src/designspace.rs

```diff
@@ -0,0 +1,213 @@
+//! Reading and writing designspace files.
+
+#![deny(rustdoc::broken_intra_doc_links)]
+
+use quick_xml::de::from_reader;
+use std::{fs::File, io::BufReader, path::Path};
+
+use crate::error::DesignSpaceLoadError;
+
+/// A [designspace]].
+///
+/// [designspace]: https://fonttools.readthedocs.io/en/latest/designspaceLib/index.html
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "designspace")]
+pub struct DesignSpaceDocument {
+    /// Design space format version.
+    pub format: f32,
+    /// One or more axes.
+    pub axes: Axes,
+    /// One or more sources.
+    pub sources: Sources,
+    /// One or more instances.
+    pub instances: Instances,
+}
+
+/// https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#axes-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "axes")]
+pub struct Axes {
+    /// One or more axis definitions.
+    pub axis: Vec<Axis>,
+}
+
+/// A [axis]].
+///
+/// [axis]: https://fonttools.readthedocs.io/en/latest/designspaceLib/xml.html#axis-element
+#[derive(Clone, Debug, Default, PartialEq, Deserialize)]
+#[serde(rename = "axis")]
+pub struct Axis {
+    /// Name of the axis that is used in the location elements.
+    pub name: String,
+    /// 4 letters. Some axis tags are registered in the OpenType Specification.
+    pub tag: String,
```

Comment:
Maybe we should be using `font-types` in norad now?

Reply:
If we do that I'd prefer it be in it's own PR. Note that currently those types don't support serde and I'm not sure they should - unless it can be optional - because we want low dependencies for users of read-fonts. I tried to use something akin to Tag in IR and found it surprisingly unpleasent to get it to interact nicely with both toml and bincode IO because for toml in particular you want to see the string tag. That may simply have been a noob error.

Reply:
Tracking in https://github.com/linebender/norad/issues/276